src/_rtmidi.pyx

```diff
@@ -401,6 +447,34 @@ cdef class MidiBase:
             self._port = None
         self.baseptr().closePort()
 
+    def set_error_callback(self, func, data=None):
+        """Register a callback function for errors.
+
+        The callback function is called when an error occurs and must take
+        three arguments. The first argument is a member of enum
+        ``RtMidiError::Type``, represented by one of the ERRORTYPE_* constants.
+        The second argument is an error message. The third argument is the
+        value of the ``data`` argument passed to this function when the
+        callback is registered.
+
+        Registering an error callback function replaces any previously
+        registered error callback.
+
+        """
+        self._error_callback = (func, data, self._decode_string)
+        self.baseptr().setErrorCallback(&_cb_error_func,
+                                        <void *>self._error_callback)
+
+    def cancel_error_callback(self):
+        """Remove the registered callback function for errors.
+
+        This can be safely called even when no callback function has been
+        registered.
+
+        """
+        self.baseptr().setErrorCallback(NULL, NULL);
+        pass
```

Comment:
No, you're right; this is absolute garbage. Normally something I'd *amend* and force push, if there's nobody looking :)